metrics/internalrequest.go

```diff
@@ -0,0 +1,91 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package metrics
+
+import (
+	"strconv"
+
+	"github.com/prometheus/client_golang/prometheus"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/metrics"
+)
+
+var (
+	InternalRequestAttemptConcurrentTotal = prometheus.NewGauge(
+		prometheus.GaugeOpts{
+			Name: "internal_request_attempt_concurrent_requests",
+			Help: "Total number of concurrent InternalRequest attempts",
+		},
+	)
+
+	InternalRequestAttemptDurationSeconds = prometheus.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Name:    "internal_request_attempt_duration_seconds",
+			Help:    "InternalRequest durations from the moment the InternalRequest PipelineRun was created til the InternalRequest is marked as finished",
+			Buckets: []float64{60, 150, 300, 450, 600, 750, 900, 1050, 1200, 1800, 3600},
```

Comment:
These numbers make sense in the context of the release-service, but are we expecting 1 hour long requests? Don't think so.

Reply:
Good question. So far I noticed an average of 5 minutes processing on IIB side. I will ask them the average build time and worst case scenario we should expect on their side.

Reply:
This is not only for IIB, but I don't think 1 hour request is that normal... Maybe it is though. @scoheb Can you shred some light here? What internal services could we be reaching that have a long time to run? Brew came to my head, but we won't use it.

Reply:
@davidmogar I discovered last week that a request will time out after 1h on IIB  - and Yes, it took that long due to a big queue on their side. So I think we are safe to keep the numbers.

Reply:
I still don't think so. Most of the requests will take less than a minute, which is the minimum value/bucket. Better values, in my opinion, would be something like `10, 20, 40, 60, 150, 300, 450, 900, 1800, 3600`. With those we get more value from the first buckets.

Reply:
IIB operations takes at least 5 minutes to finish.

Reply:
This is not an IIB operator. Also, one particular IIB operation takes that time to finish. I doubt that IIB get endpoints take, all of them, that much. Feel free to adjust the buckets, but we don't need precision for requests longer than let's say 10 minutes.

Reply:
I think the numbers you suggested look better. And Indeed, probably other endpoints other than IIB might behave very different.

*Just for the record* - I would say that this IIB operation in particular (we were using to test) is one of the fastest possible build as it contains only a single FBC fragment. This could be considerably worse depending on the number of components it has to build. But yes, this is a generic, not an IIB operator.